formwidgets/ModelObject.php

```diff
@@ -13,26 +14,42 @@ class ModelObject extends FormWidgetBase {
 
     public $emptyOption;
 
+    public ?string $searchEndpoint = null;
+
     public function init() {
         $this->fillFromConfig([
             'modelClass',
             'nameFrom',
             'emptyOption',
+            'searchEndpoint'
         ]);
         parent::init();
     }
 
+    public function loadAssets()
+    {
+        $assetsPath = '/plugins/grrramsterdam/flexiblecontent/formwidgets/modelobject/assets';
+        $this->addJs($assetsPath ."/js/modelobject.js", [
+            "build" => "Grrr.FlexibleContent",
+            "defer" => true,
+        ]);
+    }
+
     public function render() {
         $this->vars['id'] = $this->getId();
         $this->vars['name'] = $this->getFieldName();
         $this->vars['value'] = $this->getLoadValue();
         $this->vars['options'] = $this->getFieldOptions();
         $this->vars['emptyOption'] = $this->emptyOption;
+        $this->vars['searchEndpoint'] = Backend::url($this->searchEndpoint);
 
         return $this->makePartial('fc-object-field');
     }
 
     protected function getFieldOptions() {
+        if (!$this->searchEndpoint) {
+            return [];
+        }
```

Comment:
The documentation says `searchEndpoint` is an optional enhancement when the `->all()` query is a bit too heavy-handed, but this code seems to suggest it's actually required? Otherwise you won't get any results.